torch/_inductor/config.py

```diff
@@ -123,6 +123,14 @@ def fx_graph_remote_cache_default():
 # use post-grad passes.
 pre_grad_custom_pass: Optional[Callable[[torch.fx.graph.Graph], None]] = None
 
+# Registers a custom pass to be run right before fusion in Inductor scheduler.
+pre_fusion_custom_pass: Optional[
+    Callable[
+        [List["torch._inductor.scheduler.BaseSchedulerNode"]],
+        List["torch._inductor.scheduler.BaseSchedulerNode"],
+    ]
+] = None
```

Comment:
We have custom pre_grad, joint_graph, post_grad because torch.fx.aten is a more or less stable api surface.. It feels a bit weird to expose this as an api surface. 

Would it make more sense to enable this via config or monkeypatching in test for now ?

cc @jansel @Chillee

Reply:
Or at least, use underscore, denote unstable

Reply:
yeah I feel that this config opens up the possibility of having FSDP-specific IR pass ([example](https://github.com/pytorch/pytorch/pull/130095/files#diff-829be14cf2824059fb96ed721b6da45ed0e79703d21c0e22e93db2a093415a8aR151)) without hardcoding that FSDP-specific IR pass into the scheduler.

But agreed that Inductor scheduler IR is not as stable as FX IR yet and best if users don't assume it's stable - I changed the config name to prefix with `_`  and also add a warning in the config comment, to hopefully highlight this fact.